File: Codeforces/Codeforces Round #486 (Div. 3)/B - Substrings Sort.cpp
#include <bits/stdc++.h>

using namespace std;

bool cmp(string a, string b) {
    return (b.find(a) != string::npos);
}

int main() {
    int n;
    cin >> n;
    vector<string> a(n);
    for (int i = 0; i < n; ++i) {
        cin >> a[i];
    }
    sort(a.begin(), a.end(), cmp);
    /*
    for (int i = 0; i < n; ++i) {
        cout << a[i] << "\n";
    }
    */
    for (int i = 1; i < n; ++i) {
        if (a[i].find(a[i-1]) == string::npos) {
            cout << "NO";
            return 0;
        }
    }
    cout << "YES\n";
    for (int i = 0; i < n; ++i) {
        cout << a[i] << "\n";
    }
    return 0;
}

File: Codeforces/Codeforces Round #486 (Div. 3)/B - Substrings Sort.py
n = int(input())
a = []
for i in range(n):
    a.append(input())
a.sort(key=len)
isValid = True
for i in range(1, n):
    if a[i - 1] not in a[i]:
        isValid = False
        break
if isValid:
    print("YES")
    for i in range(n):
        print(a[i])
else:
    print("NO")
